contracts/SpokePool.sol

```diff
@@ -257,7 +267,15 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         );
     }
 
-    function initializeRelayerRefund(bytes32 relayerRepaymentDistributionProof) public {}
+    // This internal method should be called by an external "initializeRelayerRefund" function that validates the
+    // cross domain sender is the HubPool. This validation step differs for each L2, which is why the implementation
+    // specifics are left to the implementor of this abstract contract.
+    function _initializeRelayerRefund(bytes32 relayerRepaymentDistributionProof) internal {
+        relayerRefunds.push();
+        uint256 relayerRefundId = relayerRefunds.length - 1;
```

Comment:
small gas optitimization: get the `uint256 relayerRefundId = relayerRefunds.length;` before doing the push operation so you dont need to do the extra subtraction operation.